src/CoreWCF.Primitives/src/CoreWCF/Channels/StreamSecurityUpgradeAcceptor.cs

```diff
@@ -11,6 +12,6 @@ protected StreamSecurityUpgradeAcceptor()
         {
         }
 
-        public abstract SecurityMessageProperty GetRemoteSecurity(); // works after call to AcceptUpgrade
+        public abstract Task<SecurityMessageProperty> GetRemoteSecurityAsync(); // works after call to AcceptUpgrade
     }
```

Comment:
Keep `GetRemoteSecurity()` and add new api as virtual which calls GetRemoteSecurity. See main conversation explanation.

Reply:
I removed the GetRemoteSecurityAsync API I intoduced before as you investigated and noticed this API does not need to be async.